src/files/files.service.ts

```diff
@@ -1,79 +1,90 @@
+    async uploadFile(file: Express.Multer.File): Promise<FileDto> {
+        const objectKey = `${uuid()}-${file.originalname}`;
         try {
-            const s3 = new S3();
-            const uploadResult = await s3
-                .upload({
-                    Bucket: this.bucketName,
-                    Body: dataBuffer,
-                    Key: `${uuid()}-${filename}`,
-                })
-                .promise();
-            return { key: uploadResult.Key, url: uploadResult.Location };
+            const preSignedUrl = await firstValueFrom(
+                this.fileClient.send(
+                    { cmd: 'get_uploadUrl' },
+                    { objectKey, ContentType: file.mimetype },
+                ),
+            );
+            await axios.put(preSignedUrl, file.buffer);
+            return {
+                key: objectKey,
+                url: preSignedUrl,
+            };
         } catch (err) {
             if (err) {
-                throw new NotFoundException('Image object not found');
+                throw new BadRequestException('Failed to upload image');
             }
         }
     }
 
-    async deleteFile(key: string): Promise<FilesEntity> {
+    async downloadFile(objectKey: string): Promise<Buffer> {
         try {
-            const s3 = new S3();
-            await s3
-                .deleteObject({
-                    Bucket: this.bucketName,
-                    Key: key,
-                })
-                .promise();
-            const file = await this.filesRepo.findOneByOrFail({ key });
-            return this.filesRepo.remove(file);
+            const preSignedUrl = await firstValueFrom(
+                this.fileClient.send({ cmd: 'get_downloadUrl' }, objectKey),
+            );
+            const axiosResponse = await axios.get(preSignedUrl, {
+                responseType: 'arraybuffer',
+            });
+            return Buffer.from(axiosResponse.data, 'binary');
         } catch (err) {
             if (err) {
-                throw new NotFoundException('File or Image object not  found');
+                throw new BadRequestException(`Filed to download image`);
             }
         }
     }
 
-    async downloadFile(key: string): Promise<any> {
+    async deleteFile(key: string): Promise<FilesEntity> {
+        const file = await this.filesRepo.findOne({ where: { key } });
+        if (!file) {
+            throw new NotFoundException('Image not found');
+        }
         try {
-            const s3 = new S3();
-            return s3
-                .getObject({
-                    Bucket: this.bucketName,
-                    Key: key,
-                })
-                .promise();
+            const preSignedUrl = await firstValueFrom(
+                this.fileClient.send({ cmd: 'get_deleteUrl' }, key),
+            );
+            await axios.delete(preSignedUrl);
+            return this.filesRepo.remove(file);
```

Comment:
Probably you need to use await here to make `try/catch` possibility to handle an exception from thge `this.filesRepo.remove()` function.